presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSource.java

```diff
@@ -529,9 +529,7 @@ public CoercionLazyBlockLoader(Block block, Function<Block, Block> coercer)
         @Override
         public void load(LazyBlock lazyBlock)
         {
-            if (block == null) {
-                return;
-            }
+            requireNonNull(block, "Already loaded");
```

Comment:
These should be
```java
checkState(block != null, "Already loaded");
```
Since NPE is the wrong exception type

Reply:
i didn't use `checkState(block != null, "Already loaded")` because it causes Intellij to mark next use of `block` as "potential NPE".
I don't like to adjust code for IDE's needs, but I also don't like false warning in the code. This makes seeing real warnings harder.

Reply:
It doesn't do that for me.  Maybe you imported the wrong `checkState`. We use this in many places to ensure valid states.